File: backend/src/main/java/com/interview/platform/mod10_interview_session/repository/InterviewSessionRepository.java
package com.interview.platform.mod10_interview_session.repository;

import com.interview.platform.mod10_interview_session.model.InterviewSessionEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import java.util.UUID;

public interface InterviewSessionRepository extends JpaRepository<InterviewSessionEntity, UUID> {
    java.util.List<InterviewSessionEntity> findAllByStatus(String status);
    java.util.List<InterviewSessionEntity> findTop5ByOrderByStartedAtDesc();
    java.util.List<InterviewSessionEntity> findAllByCandidateUserIdOrderByStartedAtDesc(UUID candidateUserId);
    java.util.List<InterviewSessionEntity> findAllByCandidateUserEmailOrderByStartedAtDesc(String email);
}

File: android-app/app/src/main/java/com/interview/platform/domain/repository/InterviewSessionRepository.kt
package com.interview.platform.domain.repository

import com.interview.platform.domain.model.mod10_interview_session.InterviewSession
import com.interview.platform.data.model.Result
import kotlinx.coroutines.flow.Flow

interface InterviewSessionRepository {
    fun fetchAll(): Flow<Result<List<InterviewSession>>>
    suspend fun syncData(): Result<Unit>
}
